src/snapred/backend/recipe/algorithm/FitMultiplePeaksAlgorithm.py

```diff
@@ -0,0 +1,94 @@
+        crystalInfo = fitPeakIngredients.CrystalInfo
+        peakType = fitPeakIngredients.PeakType
+
+        beta_0 = instrumentState.gsasParameters.beta[0]
+        beta_1 = instrumentState.gsasParameters.beta[1]
+        FWHMMultiplierLeft = instrumentState.fwhmMultiplierLimit.minimum
+        FWHMMultiplierRight = instrumentState.fwhmMultiplierLimit.maximum
+        L = instrumentState.instrumentConfig.L1 + instrumentState.instrumentConfig.L2
+        ws = mtd[wsName]
+        # TODO: Fix this to use MantidSnapper when possible
+        # Currently MantidSnapper is unable to return the List
+        numSpec = ws.getNumberHistograms()
+        purgeAlgo = PurgeOverlappingPeaksAlgorithm()
+        purgeAlgo.initialize()
+        purgeAlgo.setProperty("InstrumentState", instrumentState.json())
+        purgeAlgo.setProperty("NumFocusGroups", str(numSpec))
+        purgeAlgo.setProperty("CrystalInfo", crystalInfo.json())
+        purgeAlgo.execute()
+        reducedList = json.loads(purgeAlgo.getProperty("OutputPeakMap").value)
+
+        ws_group = WorkspaceGroup()
+        mtd.add("fitPeaksWSGroup", ws_group)
+
+        for index in range(numSpec):
+            delDoD = instrumentState.pixelGroupingInstrumentState[index].delta_dhkl_over_dhkl
+            tTheta = instrumentState.pixelGroupingInstrumentState[index].twoThetaAverage
+            peakLimits = []
+            for peak, dspc in enumerate(reducedList[index]):
+                halfWindLeft = 2.35 * delDoD * dspc * FWHMMultiplierLeft
+                halfWindRight = 2.35 * delDoD * dspc * FWHMMultiplierRight
+                lowerLimit = dspc - halfWindLeft
+
+                beta_t = beta_0 + beta_1 / dspc**4
+                beta_d = 505.548 * L * np.sin(tTheta / 2) * beta_t
+                upperLimit = dspc + halfWindRight + (1 / beta_d)
+                peakLimits.extend([lowerLimit, upperLimit])
+
+            self.mantidSnapper.ExtractSingleSpectrum(
+                "Extract Single Spectrm...", InputWorkspace=wsName, OutputWorkspace="ws2fit", WorkspaceIndex=index
+            )
+
+            self.mantidSnapper.FitPeaks(
+                "Fit Peaks...",
+                InputWorkspace="ws2fit",
+                PeakCenters=",".join(np.array(reducedList[index]).astype("str")),
+                PeakFunction=peakType,
+                FitWindowBoundaryList=",".join(np.array(peakLimits).astype("str")),
+                OutputWorkspace=f"{wsName}_fitted_peakpositions_{index}",
+                OutputPeakParametersWorkspace=f"{wsName}_fitted_params_{index}",
+                BackgroundType="Quadratic",
+                FittedPeaksWorkspace=f"{wsName}_fitted_{index}",
+                ConstrainPeakPositions=True,
+                OutputParameterFitErrorsWorkspace=f"{wsName}_fitted_params_err_{index}",
+            )
+            self.mantidSnapper.executeQueue()
+            ws_group.add(f"{wsName}_fitted_peakpositions_{index}")
```

Comment:
Can these names be declared as constants somewhere earlier?

Reply:
These names are the names of the output workspaces created by `FitPeaks` the workspace name is provided by the user and the index corresponds to the number of histograms in the workspace. I could create a list of ws names prior to calling `FitPeaks` but I don't see how that would be beneficial?

Reply:
To Reece's point, you can declare the template strings earlier and format them for the inputs here.

some psuedoish code:
```
workspaceNameTemplate = "{wsName}_fitted_params_err_{index}"
...
    OutputParameterFitErrorsWorkspace=workspaceNameTemplate.format(wsName=wsName, index=index)",
)
```

Reply:
That's what I meant, since you use each name twice.
```
ws_fittted_peaks = f"{wsName}_fitted_peakpositions_{index}"
ws_fitted_params = f"{wsName}_fitted_params_{index}"
ws_fitted = f"{wsName}_fitted_{index}"
ws_fitted_params_err = f"{wsName}_fitted_params_err_{index}"
 
 FitWindowBoundaryList=",".join(np.array(peakLimits).astype("str")),
                OutputWorkspace=ws_fitted_peaks,
                OutputPeakParametersWorkspace=ws_fitted_params,
                BackgroundType="Quadratic",
                FittedPeaksWorkspace=ws_fitted,
                ConstrainPeakPositions=True,
                OutputParameterFitErrorsWorkspace=ws_fitted_params_err,
            )
 ws_group.add(ws_fitted_peaks)
 ws_group.add(ws_fitted_params)
 ws_group.add(ws_fitted)
 ws_group.add(ws_fitted_params_err)
```